### PR TITLE
DSD-920: Updating styles for the datepicker's calendar popup

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-920](https://jira.nypl.org/browse/DSD-920)

## This PR does the following:

- Updates the styles for `React Datepickers`'s calendar popup used in the `DatePicker` component.

## How has this been tested?

Storybook.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

**Comment by bigfishdesign13** [comment]
@EdwinGuzman Take a look at this on Mac Safari.  The prev/next arrows are not visible.  But this was not due to anything you added.  Apparently, it has always been like this.  It is from the positioning attributes.  If you fiddle with the inspect tools in the browser, you can eventually get them to show.  But other than that, I don’t know exactly which styles are causing this.

image

**Comment by EdwinGuzman** [comment]
It looks like this is an existing bug in prod: https://nypl.github.io/nypl-design-system/storybook-static/?path=/story/components-form-elements-datepicker--date-picker-with-controls

Edit: Never mind, you mentioned that. Are there any other elements that have some issue in Safari?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

**Comment by EdwinGuzman** [comment]
Updated and fixed.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

File: src/components/DatePicker/_DatePicker.scss [hunk 1]
```diff
@@ -1,42 +1,93 @@
...
       border-radius: 0;
-      color: var(--nypl-colors-ui-white);
+      color: var(--nypl-colors-ui-black);
       opacity: 0.3;
     }
-
+    // For today's date visual styles.
+    &-text--today,
+    &--today {
+      color: var(--nypl-colors-ui-black);
```
Comment (bigfishdesign13) [inline src/components/DatePicker/_DatePicker.scss:79]: The basic hover styles are overwriting these styles.  While it’s not a best practice, I think the color and BG color attributes here need to be locked in using `!important`.

File: src/components/DatePicker/_DatePicker.scss [hunk 2]
```diff
@@ -1,42 +1,93 @@
...
+      background-color: var(--nypl-colors-ui-gray-light-cool);
+    }
     // Selected states.
     &-text--selected,
     &--selected {
-      background-color: var(--nypl-colors-ui-black);
+      background-color: var(--nypl-colors-ui-focus);
```
Comment (bigfishdesign13) [inline src/components/DatePicker/_DatePicker.scss:89]: Same thing here.  The basic hover styles are overwriting these styles.  The color and BG color attributes here need to be locked in using `!important`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The arrows look good, but some of the hover styles need to be tweaked.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

File: src/components/DatePicker/_DatePicker.scss [hunk 3]
```diff
@@ -1,42 +1,101 @@
...
     &-text--keyboard-selected,
     &--keyboard-selected,
     &--in-range,
     &--in-selecting-range {
-      background-color: var(--nypl-colors-ui-black);
+      background-color: var(--nypl-colors-ui-gray-light-cool);
       border-radius: 0;
-      color: var(--nypl-colors-ui-white);
+      color: var(--nypl-colors-ui-black);
       opacity: 0.3;
```
Comment (bigfishdesign13) [inline src/components/DatePicker/_DatePicker.scss]: This is causing trouble and I think it should be removed.  Can you find a usecase where it makes sense?

File: src/components/DatePicker/_DatePicker.scss [hunk 4]
```diff
@@ -1,42 +1,101 @@
...
+      opacity: 1;
     }
-
     // For ranges about to be selected.
     &-text--keyboard-selected,
     &--keyboard-selected,
     &--in-range,
     &--in-selecting-range {
-      background-color: var(--nypl-colors-ui-black);
+      background-color: var(--nypl-colors-ui-gray-light-cool);
```
Comment (bigfishdesign13) [inline src/components/DatePicker/_DatePicker.scss]: Becuase I am looking at what’s happening with the code on line 75, I can now see that this should be set to `var(--nypl-colors-ui-gray-x-light-cool)`

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Ugh.  One more change and one question.  Also, the `minDate` and `maxDate` on the `DatePicker` should be updated to more recent dates.

**Comment by EdwinGuzman** [comment]
Regarding the last round of feedback, it was mostly because of the dates within a range of selected dates.
Screen Shot 2022-04-19 at 2 53 04 PM

In this example, the "from" date is 4/13, today's date is 4/19, and the "to" date is 4/22. When I go to select 4/4, all the dates from 4 to 12 look the same as the dates within the selected dates of 4/13 to 4/22. So the opacity was there for that. But from the screenshot above, the same gray x-light-cool looks fine. What do you think?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

**Review by bigfishdesign13 — APPROVED** [review]
The coloring in the example you showed looks good.  Thank you for working through these issues.

**Comment by EdwinGuzman** [comment]
No problem!

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr959-wk6u9rz2qzvsmmglcfa9t6q47o5l6i7w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6259c8adbda4981539fd8288

**Pull request merged**